knip.jsonc

```diff
@@ -1,5 +1,5 @@
 {
 	"$schema": "https://unpkg.com/knip@next/schema.json",
-	"entry": "src/index.ts!",
-	"project": "src/**/*.ts!"
+	"entry": ["src/index.ts!", "src/**/*.test.ts"],
```

Comment:
I'm confused - I wouldn't think it right to include test files as entry points. Then any unused code that is tested wouldn't be detected.

Can you talk more about how+why this is useful? E.g. should we be running Knip in _two_ modes: production and test?

Reply:
> any unused code that is tested wouldn't be detected.

When running knip in production mode, test files won't be entry points and thus we'll detect this case then.
When not in production mode, we'd want to detect code not used at all. So production code, but also test helper function such as those in `src/test/utils.ts`.

> should we be running Knip in _two_ modes: production and test?

Yes, I think so.

**Caveat:**

I haven't use knip before, these are just my initial thoughts on how it should work.

Reply:
Without `"src/**/*.test.ts"` as an entry point and with `"src/**/*.ts"` as the project files, I would of thought that `createNode` would be flagged as not used, as it's only imported by test files. This don't seem to be the case though, so I'm not entirely sure how knip determines not used code.

Reply:
Ah cool thanks that makes sense! CCing @webpro in case you have time to weigh in here, since you've already been helpful on this repo 😄 (no worries if not!)

Reply:
We might want to also exclude `src/test` from production project files:

```json
"project": [
	"src/!(test)/**/!(*.test).ts!",
	"src/!(*.test).ts!",
	"src/**/*.ts"
]
```